plugin/pkg/admission/namespace/exists/admission.go

```diff
@@ -41,10 +39,12 @@ func init() {
 // It is useful in deployments that want to enforce pre-declaration of a Namespace resource.
 type exists struct {
 	*admission.Handler
-	client clientset.Interface
-	store  cache.Store
+	client          clientset.Interface
+	informerFactory informers.SharedInformerFactory
```

Comment:
When you sweep for `HasSynced` go ahead update these to hold only the information they need.  You still init based on the factory, but you store the values you need.